autotest.pm

```diff
@@ -471,4 +474,25 @@ sub loadtestdir ($dir) {
     }
 }
 
+# this is called if the framework loaded a VM snapshot - all consoles
+# activated in the test's run loose their state but we should not
+# reset consoles which where already activated
+# in the last milestone.
+sub rollback_activated_consoles () {
+
+    for my $console (@$activated_consoles) {
+        next if (grep { /^$console$/ } @$last_milestone_active_consoles);
+        autotest::query_isotovideo('backend_reset_console', {testapi_console => $console});
+    }
+    $activated_consoles = [];
+
+    if (defined($last_milestone_console)) {
+        my $ret = autotest::query_isotovideo('backend_select_console',
+            {testapi_console => $last_milestone_console});
+        die $ret->{error} if $ret->{error};
+    }
+
+    return;
```

Comment:
```suggestion
    for my $console (@$activated_consoles) {
        next if grep { /^$console$/ } @$last_milestone_active_consoles;
        autotest::query_isotovideo('backend_reset_console', {testapi_console => $console});
    }
    $activated_consoles = [];
    if (defined($last_milestone_console)) {
        my $ret = autotest::query_isotovideo('backend_select_console',
            {testapi_console => $last_milestone_console});
        die $ret->{error} if $ret->{error};
    }
    return;
```

Reply:
you don't like free lines between "logical" sections inside a function?

Reply:
regarding 2)
is it about this line:
```
3: CLEMIX cmd: backend_reset_consoleCLEMIX cmd: backend_select_console[09:35:55] t/08-autotest.t .......................... ok     1822 ms ( 0.02 usr  0.00 sys +  1.62 cusr  0.20 csys =  1.84 CPU)
```
This was a leftover :facepalm:  sorry!

Reply:
You got me there :) blank lines between logical sections inside a function are an indication that the logical sections maybe should be their own functions. As alternative reconsider to see that the separation between the logical sections shouldn't be *that* big and just remove the blank lines ;)

Reply:
I think this function is small enough and I'd only remove the blank line at the beginning. I like the reduced use of brackets in @okurz suggestion, though.